projects/auth0-angular/src/lib/auth.service.spec.ts

```diff
@@ -199,6 +199,46 @@ describe('AuthService', () => {
         done();
       });
     });
+
+    it('should record errors in the error$ observable', (done) => {
+      const errorObj = new Error('An error has occured');
+
+      (auth0Client.handleRedirectCallback as jasmine.Spy).and.throwError(
+        errorObj
+      );
+
+      const localService = createService();
+
+      loaded(localService).subscribe(() => {
+        localService.error$.subscribe((err: Error) => {
+          expect(err).toBe(errorObj);
+          expect(navigator.navigateByUrl).toHaveBeenCalledWith('/');
+          done();
+        });
+      });
+    });
+
+    it('should process the callback when an error appears in the query string', (done) => {
+      const errorObj = new Error('An error has occured');
+
+      (auth0Client.handleRedirectCallback as jasmine.Spy).and.throwError(
```

Comment:
comparing these two tests, the only difference is that this second one is changing the "search" (query) value. But how can I know that that is actually having some influence in what we're testing? e.g. remove that line and the test is the same as the above one.

I think this line I'm commenting on should probably be gone, as the error should be parsed and raised by the auth service rather than us mocking that.

Reply:
Yeah, the problem is that `handleRedirectCallback` is completely mocked (it's a method on the SPA SDK). The fact that the test passes with those query string values show that the check for whether a callback should be handled actually works.

I'm not sure how else to test that, but open to suggestions.

Reply:
I've updated this test now, as discussed.